autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/sorting/CredentialListSorter.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.ui.credential.management.sorting
+
+import com.duckduckgo.autofill.AutofillDomainFormatter
+import com.duckduckgo.autofill.domain.app.LoginCredentials
+import com.duckduckgo.di.scopes.FragmentScope
+import com.squareup.anvil.annotations.ContributesBinding
+import java.text.Collator
+import javax.inject.Inject
+
+interface CredentialListSorter {
+    fun sort(credentials: List<LoginCredentials>): List<LoginCredentials>
+    fun comparator(): Collator
+}
+
+@ContributesBinding(FragmentScope::class)
+class CredentialListSorterByTitleAndDomain @Inject constructor(
+    private val domainFormatter: AutofillDomainFormatter
+) : CredentialListSorter {
+
+    override fun sort(credentials: List<LoginCredentials>): List<LoginCredentials> {
+        return credentials.sortedWith(credentialComparator)
+    }
+
+    override fun comparator(): Collator {
+        return buildCollator()
```

Comment:
do we need a new instance of the collator here and the credentialComparator? I am not very familiar with this class but checking if they can't be reused.

Reply:
Not 100% sure but I know it carries some state so happy to give each use case its own instance